CMakeLists.txt

```diff
@@ -205,6 +205,49 @@ else()
   endif()
 endif()
 
+message(INFO " HAVE_ROCSPARSE: ${HAVE_ROCSPARSE}")
+find_package(hip)
+
+message(INFO " hip version: ${hip_VERSION}")
+message(INFO " rocblas version: ${rocblas_VERSION}")
+message(INFO " rocsparse version: ${rocsparse_VERSION}")
+message(INFO " CMAKE_CXX_COMPILER: ${CMAKE_CXX_COMPILER}")
+message(INFO " HAVE_ROCSPARSE: ${HAVE_ROCSPARSE}")
+
+if(hip_FOUND)
+  #message(INFO " CXX_COMPILER: ${CXX_COMPILER}")
+  get_filename_component(CXX_COMPILER ${CMAKE_CXX_COMPILER} NAME)
+  #message(INFO " CXX_COMPILER: ${CXX_COMPILER}")
+  message(INFO " CMAKE_CXX_COMPILER_ID: ${CMAKE_CXX_COMPILER_ID}")
+  if(hip_VERSION VERSION_LESS "5.3")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      message(WARNING " Cannot use gcc for rocm < 5.3")
+      unset(HAVE_ROCSPARSE)
+    endif()
+    find_package(rocalution)
+    if(ROCALUTION_FOUND)
+      set(HAVE_ROCALUTION 1)
+      set(COMPILE_BDA_BRIDGE 1)
+    endif()
+  else()
+    find_package(rocblas)
+    find_package(rocsparse)
+    if(rocsparse_FOUND AND rocblas_FOUND)
+      set(HAVE_ROCSPARSE 1)
+      set(COMPILE_BDA_BRIDGE 1)
+    endif()
```

Comment:
Even without rocblas HAVE_ROCSPARSE is 1 on my system before the if clause. Hence we need an else branch that unsets this.

Generally, I am wondering whether rocsparse really needs rocblas. Are you sure about this? Or is it just that our code will use it.

On my system:
```
ldd /usr/lib/x86_64-linux-gnu/librocsparse.so
	linux-vdso.so.1 (0x00007ffcce948000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f45892d7000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f45891f8000)
	libamdhip64.so.5 => /lib/x86_64-linux-gnu/libamdhip64.so.5 (0x00007f453a200000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4539e00000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f453a01f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f4589315000)
	libhsa-runtime64.so.1 => /lib/x86_64-linux-gnu/libhsa-runtime64.so.1 (0x00007f4539a00000)
	libnuma.so.1 => /lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f45891e9000)
	libhsakmt.so.1 => /lib/x86_64-linux-gnu/libhsakmt.so.1 (0x00007f453afd5000)
	libelf.so.1 => /lib/x86_64-linux-gnu/libelf.so.1 (0x00007f45891ce000)
	libdrm.so.2 => /lib/x86_64-linux-gnu/libdrm.so.2 (0x00007f453afbf000)
	libdrm_amdgpu.so.1 => /lib/x86_64-linux-gnu/libdrm_amdgpu.so.1 (0x00007f453afb3000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f4539de1000)
```

Reply:
It seems rocsparse does not depend on rocblas, but our solver does, due to the dense vector operations (axpy, etc.).